### ISSUE TITLE
pouchdb-browser fails importing from 'events'

### ISSUE THREAD

**Issue by jdfwarrior** [pouchdb/pouchdb#8641]
### Issue
I just started on my project and still haven't actually used pouchdb at all yet. I'm in the middle of trying to upgrade a bunch of dependencies that are a few years old and don't even have the app running yet. One such update was migrating from webpack to vite as the build tool and I've come across a few issues.

We're using pouchdb-browser.

As defined in the documentation under the plugins section, says that pouchdb comes default with the pouchdb-replication plugin. pouchdb-replication however imports EE from 'events' which isn't available in the browser. It's throwing an error for me when the Changes class attempts to extend EE, because EE is null or not a constructor. Maybe this could be resolved by replacing imports for 'events' with something like 'eventemitter3'.

I also initially ran into an issue where the `immediate` lib (one of pouchdb-browser's dependencies) was throwing an error because the `mutation` lib within it attempts to initialize a `Mutation` variable to `global.MutationObserver` and vite doesn't provide shims for node like webpack does (reference: https://github.com/vitejs/vite/issues/2778#issuecomment-810086159). This can be resolved by importing a module that sets `window.global = window` and make it your first import. I know this is outside the scope of pouchdb, but if someone is running into these issues, they're potentially doing something similar and wanted to document the cause.

### Info
- Environment: browser
- Platform: electron

### Reproduce
I didn't go make a base case for this but I would think it should be easily reproducible by creating a new vite project and importing pouchdb-browser

**Comment by jatin-2708** [comment]
Please assign me this issue, very much excited to mark may first contribution to this.

**Comment by twitwi** [comment]
Thanks for the issue and the work in progress.
If anyone has the problem with import_events in the browser, I managed to work around it thanks to this issue, with `npm install events`.
And for the missing global, I updated my vite config

```
  define: { // https://stackoverflow.com/questions/72114775/vite-global-is-not-defined
    global: 'window',
  },
```

**Comment by alxndrsn** [comment]
> I also initially ran into an issue where the `immediate` lib (one of pouchdb-browser's dependencies) was throwing an error because the `mutation` lib within it attempts to initialize a `Mutation` variable to `global.MutationObserver`

This may be caused by https://github.com/calvinmetcalf/immediate/issues/39.

**Comment by zaynekomichi** [comment]
> Thanks for the issue and the work in progress. If anyone has the problem with import_events in the browser, I managed to work around it thanks to this issue, with `npm install events`. And for the missing global, I updated my vite config
> 
> ```
>   define: { // https://stackoverflow.com/questions/72114775/vite-global-is-not-defined
>     global: 'window',
>   },
> ```

This solved it for me, seems the events package is needed

**Comment by garethbowen** [comment]
Fixed by #8849
